### PR TITLE
#2044 Add Ignore errors and timeout for ORM Cache

### PR DESCRIPTION
**Pull request by dheepak-aot**
## Ignore Errors and add timeout 
- [x] Added `ignoreErrors: true` to the orm cache config.
 - [x] Set `commandTimeout` of redis to 60 seconds in cache config which is infinite by default.

## Retry Strategy (https://github.com/redis/ioredis/tree/v4#auto-reconnect)
When `ignoreErrors: true` is set, errors are ignored and the typeorm works normally but the redis client(ioredis and ioredis/cluster) does retry every 2 seconds to connect again to redis server and creates log for every retry.

This is due to the default implementation of `retryStrategy` and `clusterRetryStrategy`.

image

image

- [x] Hence the connection retry interval has been set to 60 seconds for orm config.

### CODE REVIEW

**Review by andrewsignori-aot — APPROVED** [review]
Nicely done and thanks for the further explanations on the PR description. Looks good 👍

File: sources/packages/backend/libs/utilities/src/system-configurations-constants.ts [hunk 1]
```diff
@@ -24,6 +24,22 @@ export const MIN_CANADA_LOAN_OVERAWARD = 250;
...
+/**
+ * Redis retry interval to retry connection
+ * when a connection cannot be established to redis.
+ *
+ * Ref for standalone: https://github.com/redis/ioredis/tree/v4#auto-reconnect
+ *
+ * Ref for cluster: https://github.com/redis/ioredis/tree/v4#cluster
+ */
+export const ORM_CACHE_REDIS_RETRY_INTERVAL = 60 * 1000;
+
```
Comment (guru-aot) [inline sources/packages/backend/libs/utilities/src/system-configurations-constants.ts:42]: For my understanding. Why are we having the redis retry and the timeout same interval? Doesnt the retry should be lesser or the timeout more?

File: sources/packages/backend/libs/sims-db/src/data-source.ts [hunk 2]
```diff
@@ -100,7 +104,10 @@ function getORMCacheConfig(): ORMCacheConfig | false {
         host: config.redis.redisHost,
         port: config.redis.redisPort,
         password: config.redis.redisPassword,
+        commandTimeout: ORM_CACHE_REDIS_COMMAND_TIMEOUT,
+        retryStrategy: () => ORM_CACHE_REDIS_RETRY_INTERVAL,
       },
+      ignoreErrors: true,
```
Comment (guru-aot) [inline sources/packages/backend/libs/sims-db/src/data-source.ts:110]: 👍

**Review by ann-aot — APPROVED** [review]
👍

**Review by sh16011993 — APPROVED** [review]
LGTM 👍

**Review by guru-aot — APPROVED** [review]
LGTM,  nice work @dheepak-aot

**Pull request merged**